feeds/product.php

```diff
@@ -38,6 +38,151 @@
 
 DfTools::validateSecurityToken(Tools::getValue('dfsec_hash'));
 
+/**
+ * Global storage for all active measurements.
+ *   - $__metrics_data[$label] holds ['time' => float, 'memory' => int, 'peak' => int]
+ *   - $__metrics_stack is an indexed array of labels, used to pop the most recent.
+ */
+$__metrics_data  = [];
+$__metrics_stack = [];
+
+/**
+ * Begin measuring time & memory.
+ *
+ * @param string|null $label  Optional: a custom label for this measurement.
+ *                            If you pass null (or omit), a unique label is generated.
+ * @return string             The label that was used.  (Use the same label in stop_metrics() if you want.)
+ */
+function start_metrics(string $label = null): string
+{
+    global $__metrics_data, $__metrics_stack;
+
+    // 1) If no label provided, generate one
+    if ($label === null) {
+        // uniqid('m_', true) gives something like "m_5cd1e4d3a1f71.12345600"
+        $label = uniqid('m_', true);
+    }
+
+    // 2) Warn if the same label is already running
+    if (isset($__metrics_data[$label])) {
+        trigger_error("start_metrics(): metrics with label '$label' already started.", E_USER_WARNING);
+    }
+
+    // 3) Capture "before" values
+    $t0 = microtime(true);
+    $m0 = memory_get_usage();
+    $p0 = memory_get_peak_usage();
+
+    // 4) Store them under $__metrics_data
+    $__metrics_data[$label] = [
+        'time'   => $t0,
+        'memory' => $m0,
+        'peak'   => $p0,
+    ];
+
+    // 5) Push this label onto the stack (so stop_metrics() with no args can pop it)
+    $__metrics_stack[] = $label;
+
+    return $label;
+}
+
+/**
+ * Stop measuring time & memory and compute deltas.
+ *
+ * @param string|null $label  Optional: the label you used in start_metrics().
+ *                            If you pass null, this function will pop the most recently started label.
+ * @return array|null         On success, returns an array:
+ *                                [
+ *                                  'label'       => (string) the label used,
+ *                                  'time'        => (float) elapsed seconds,
+ *                                  'memory'      => (int)   bytes of memory allocated,
+ *                                  'peak_memory' => (int)   extra peak memory (bytes)
+ *                                ]
+ *                            On failure (no matching start), returns null and emits a warning.
+ */
+function stop_metrics(string $label = null): ?array
+{
+    global $__metrics_data, $__metrics_stack;
```

Comment:
Check this against the PrestaShop validator before merging

```suggestion
    global $__metricsData, $__metricsStack;
```